### PR TITLE
RF-12: Sign up process

### PR DESCRIPTION
**Pull request by AgustinCarthery**
# [RF-12: Sign up process](https://rutinesandfit.atlassian.net/browse/RF-12)

- Feature (new feature of the server/application)

## Description

- CreateAccount and CompleteData screens
- Yup validations
- Header component
- Firebase signup thunks

## Attachments
Screenshot 2024-01-30 at 14 43 12

Screenshot 2024-01-30 at 14 43 31

## Change Procedure

- We need to change the way we handle authentication flag on root-stack, maybe using AsyncStore with jwt when login in is a good option
- We need to implement a date picker and a selector component

### CODE REVIEW

File: src/app/navigation/auth.tsx [hunk 1]
```diff
@@ -1,14 +1,45 @@
...
+        />
+        <RootStack.Screen
+          name="CompleteData"
+          component={CompleteDataScreen}
+          options={({ navigation }) => ({
+            header: () => (
+              <Header
+                leftIcon={<BackArrowIcon />}
+                navigation={navigation}
+                headerTitle="Informacion Personal"
```
Comment (AgustinCarthery) [inline src/app/navigation/auth.tsx]: forgot about this , will fix it 😅

File: src/app/screens/auth/signup/completeData/index.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      birthDate: '',
+      gender: undefined,
+    },
+    resolver: yupResolver(validationSchema),
+  });
+
+  const onValidSubmit: SubmitHandler<FormData> = async data => {
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { firstName, lastName, birthDate, gender } = data;
```
Comment (AgustinCarthery) [inline src/app/screens/auth/signup/completeData/index.tsx]: im going to enable the rule once we dispatch the create user info action

**Review by franco-marini — COMMENTED** [review]
- Review files with kebab-case
- Move the types inside a type file and re-use
- Form config must have

**Review by franco-marini — APPROVED** [review]
🚀

**Pull request merged**